src/Template/Element/advanced_search_form.ctp

```diff
@@ -29,6 +29,21 @@ echo $this->Form->create('AdvancedSearch', [
 ]);
 ?>
 
+<?php if ($this->request->getParam("?")): ?>
+    <md-divider>
+        <md-button class="md-primary md-raised"
+                    href="<?= $this->Url->build([
+                        "controller" => "Sentences",
+                        "action" => "advanced_search",
+                        "?" => [],
+                    ]);?>">
+            <?= __('Clear') ?>
+        </md-button>
+        <i><?= __("This form has been filled by a template.")?></i>
+    </md-divider>
+    <md-divider></md-divider>
+<?php endif; ?>
```

Comment:
This block is visible as unstyled text during page load. We can use the HTML attribute `ng-cloak` to [prevent this](https://www.wikitechy.com/angularjs/directives/angularjs-ngcloak). So I suggest you move this code inside the div below `<div layout="column" ng-app="app" ng-cloak>`.